File: move-zeros/solutions.py
from typing import List


class Solution:
    def move_zeros(self, nums: List[int]):
        n = len(nums)
        fast = slow = 0
        while fast < n:
            if nums[fast] != 0:
                nums[slow], nums[fast] = nums[fast], nums[slow]
                slow += 1
            fast += 1
# 解法思路 - 双指针
# 快指针在遇到不为0的数，和慢指针交换值，然后慢指针进1，快指针进1

File: move-zeros/solutions.ts
function moveZeros(nums: number[]): void {
  let slow = 0;
  for (let fast = 0; fast < nums.length; fast++) {
    if (nums[fast] !== 0) {
      nums[slow] = nums[fast];
      slow++;
    }
  }
  nums.fill(slow, 0);
}
// 快指针一直走，遇到不为零的交换到慢指针，慢指针进1，最后fill覆盖填0
